assembl/static2/js/app/components/debate/thread/nuggets.jsx

```diff
@@ -4,8 +4,8 @@ import { getDomElementOffset, computeDomElementOffset } from '../../../utils/glo
 
 class Nuggets extends React.Component {
   static nodeExtracts(node) {
-    let result = node.getElementsByClassName('posts')[0];
-    if (result) result = result.getElementsByClassName('extracts')[0];
+    let result = Nuggets.getChildsByClassName(node, 'posts')[0];
+    if (result) result = Nuggets.getChildsByClassName(result, 'extracts')[0];
```

Comment:
Ahh okay, you used this same method as before. Yes, traversing the DOM is restrictive. What you could have done, in fact, is changed the component structure a bit. You could have passed the parent node as a prop to each nugget. And then, using `this.node.props` discovered if there is a parent, and computed the level that way. But, anyways, this isn't a bad option either. It's more jQuery-style than React style. Keep that in mind for next time.